### ISSUE TITLE
Support Gradle 8.x

### ISSUE THREAD

**Issue by msgilligan** [bitcoinj/bitcoinj#2779]
To support Gradle 8.x we need to resolve (at least) the following:

Deprecation warnings from Gradle 7.6.x (`--warning-mode all`):

1. Use `mainClass` instead of `main` in `JavaExec` 2 tasks. (See PR #2780) -- Fixing this _may_ not be necessary, I think Gradle decided to not remove this until Gradle 9.
2. `Task ':bitcoinj-core:processResources' uses this output of task ':bitcoinj-core:generateProto' without declaring an explicit or implicit dependency.`

Errors when attempting to use Gradle 8.0.x:

* `project ':bitcoinj-core'` - `unknown property 'classifier' for task ':bitcoinj-core:javadocJar' of  type org.gradle.api.tasks.bundling.Jar` (Fixed by either PR #3057 or PR #3058)

**JDK 20** requires Gradle 8 or later.

**Comment by msgilligan** [comment]
The complete error message for deprecation warning (2) is:
```
> Task :bitcoinj-core:processResources
Execution optimizations have been disabled for task ':bitcoinj-core:processResources' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/Users/sean/git/bitcoinj/core/src/main/resources'. Reason: Task ':bitcoinj-core:processResources' uses this output of task ':bitcoinj-core:generateProto' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.6.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/sean/git/bitcoinj/core/src/main/proto'. Reason: Task ':bitcoinj-core:processResources' uses this output of task ':bitcoinj-core:generateProto' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.6.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
Gradle detected a problem with the following location: '/Users/sean/git/bitcoinj/core/src/main/resources'. Reason: Task ':bitcoinj-core:processResources' uses this output of task ':bitcoinj-core:generateProto' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.6.1/userguide/validation_problems.html#implicit_dependency for more details.
Gradle detected a problem with the following location: '/Users/sean/git/bitcoinj/core/src/main/proto'. Reason: Task ':bitcoinj-core:processResources' uses this output of task ':bitcoinj-core:generateProto' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.6.1/userguide/validation_problems.html#implicit_dependency for more details.
```

**Comment by msgilligan** [comment]
See PR #3061 for a fix/workaround

**Comment by msgilligan** [comment]
Fixed by PR #3059
